core/common/src/DateTimePeriod.kt

```diff
@@ -19,10 +21,29 @@ import kotlinx.serialization.Serializable
  *
  * The time components are: [hours], [minutes], [seconds], [nanoseconds].
  *
- * A `DateTimePeriod` can be constructed using the same-named constructor function,
- * [parsed][DateTimePeriod.parse] from a string, or returned as the result of instant arithmetic operations (see [Instant.periodUntil]).
- * All these functions can return a [DatePeriod] value, which is a subtype of `DateTimePeriod`,
- * a special case that only stores date components, if all time components of the result happen to be zero.
+ * ### Interaction with other entities
+ *
+ * [DateTimePeriod] can be returned from [Instant.periodUntil], representing the difference between two instants.
+ * Conversely, there is an [Instant.plus] overload that accepts a [DateTimePeriod] and returns a new instant.
+ *
+ * [DatePeriod] is a subtype of [DateTimePeriod] that only stores the date components and has all time components equal
+ * to zero.
+ *
+ * ### Construction, serialization, and deserialization
+ *
+ * When a [DateTimePeriod] is constructed in any way, a [DatePeriod] value, which is a subtype of [DateTimePeriod],
+ * will be returned if all time components happen to be zero.
+ *
+ * A `DateTimePeriod` can be constructed using the constructor function with the same name.
+ *
+ * [parse] and [toString] methods can be used to obtain a [DateTimePeriod] from and convert it to a string in the
+ * ISO 8601 extended format (for example, `P1Y2M6DT13H`).
+ *
+ * `DateTimePeriod` can also be returned as the result of instant arithmetic operations (see [Instant.periodUntil]).
+ *
+ * Additionally, there are several `kotlinx-serialization` serializers for [DateTimePeriod]:
+ * - [DateTimePeriodIso8601Serializer] for the ISO 8601 format;
+ * - [DateTimePeriodComponentSerializer]  for an object with components.
```

Comment:
> Why one might want to use `DateTimePeriod`? This is both about the usage pattern and some examples.

> a poor man's substitute to the "what it is and how to use it"

I don't get this point at all. "What it is" is answered in the first line of the documentation: it's a difference between the two instants. I wouldn't expect the documentation of `List` to start giving advice on how to find the sum of its elements, for example. It's just data, and how you want to use it is up to you.

For `LocalDateTime`, it's viable to explain all the usages we deem correct, as there are few, with the data structure being so error-prone and misunderstood. And for `Instant`, not quite.

"Interaction with other entities" lists some typical usages, but these are not the only ones. You could need to simply take `DateTimePeriod(hours = 2) + DateTimePeriod(days = 5) - 1.days.toDateTimePeriod()`, or store a `DateTimePeriod` in your program to later display it, or whatever. Since the data structure doesn't seem error-prone, we don't railroad anyone into specific usage.

Reply:
Let me rephrase my confusion a bit: I don't think "can be added to an `Instant`" is the property of `DateTimePeriod`; rather, it's the property of `Instant` that a `DateTimePeriod` can be added to it. If there were an operation like `DateTimePeriod.addToInstant(instant: Instant, timeZone: TimeZone): Instant`, then I'd agree that it not being documented in the top-level `DateTimePeriod` documentation is a mistake.

Reply:
Added the description of the overflow behavior and the meaning of the negative components.

Reply:
Also, to avoid circling back and forth on this topic, please consider this as feedback, rather than a request for change.
The overall improvement is great, and it would be a shame to bikeshedding this particular aspect in the pursuit of perfection